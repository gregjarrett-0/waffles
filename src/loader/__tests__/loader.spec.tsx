import { render } from '@testing-library/react';

import { Loader } from '../index';

describe('Loader', () => {
  it('renders a loader', () => {
    const { getByTestId } = render(
      <Loader aria-label="Basic Loader example" />,
    );
    const loader = getByTestId('loader-wrapper');

    expect(loader).toBeInTheDocument();
    expect(loader).toMatchSnapshot();
  });

  it('renders snapshot of inverted', () => {
    const { getByTestId } = render(
      <Loader inverted aria-label="Inverted Loader example" />,
    );
    const loader = getByTestId('loader-wrapper');

    expect(loader).toBeInTheDocument();
    expect(loader).toMatchSnapshot();
  });
});
