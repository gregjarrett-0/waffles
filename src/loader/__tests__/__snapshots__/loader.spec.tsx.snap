// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders a loader 1`] = `
@keyframes animation-0 {
  0%, 6% {
    -webkit-clip-path: polygon(0% -12%, 0% -12%, 169% 63%, 169% 63%);
    clip-path: polygon(0% -12%, 0% -12%, 169% 63%, 169% 63%);
  }

  100% {
    -webkit-clip-path: polygon(0% -12%, 0% 308%, 169% 383%, 169% 63%);
    clip-path: polygon(0% -12%, 0% 308%, 169% 383%, 169% 63%);
  }
}

@keyframes animation-1 {
  0%, 71% {
    -webkit-clip-path: polygon(0% 0%, 0% 78.5%, 100% 34.5%, 100% -44%);
    clip-path: polygon(0% 0%, 0% 78.5%, 100% 34.5%, 100% -44%);
  }

  96%, 100% {
    -webkit-clip-path: polygon(0% 0%, 0% 140%, 100% 96%, 100% -44%);
    clip-path: polygon(0% 0%, 0% 140%, 100% 96%, 100% -44%);
  }
}

@keyframes animation-2 {
  100% {
    stroke-dashoffset: 0;
  }
}

.emotion-0 {
  -webkit-animation: animation-0 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  animation: animation-0 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  will-change: clip-path;
}

.emotion-1 {
  -webkit-animation: animation-1 cubic-bezier(0, 0, 0.85, 1) 2s infinite alternate;
  animation: animation-1 cubic-bezier(0, 0, 0.85, 1) 2s infinite alternate;
  will-change: clip-path;
}

.emotion-2 {
  -webkit-clip-path: polygon(-0.1% -10%, 169% 65%, -0.1% 139%);
  clip-path: polygon(-0.1% -10%, 169% 65%, -0.1% 139%);
}

.emotion-3 {
  display: block;
  stroke: #000820;
  overflow: visible;
}

.emotion-4 {
  -webkit-animation: animation-2 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  animation: animation-2 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  stroke-dasharray: 9800;
  stroke-dashoffset: 9800;
  will-change: stroke-dashoffset;
}

<div
  class="emotion-0"
  data-testid="loader-wrapper"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <svg
        aria-label="Basic Loader example"
        class="emotion-3"
        viewBox="0 0 2640 3444"
      >
        <path
          class="emotion-4"
          d="M0 0 M2569 1056L143 2447V149l1175 673v1867l1248 715"
          fill="none"
          stroke-linejoin="round"
          stroke-width="300"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Loader renders snapshot of inverted 1`] = `
@keyframes animation-0 {
  0%, 6% {
    -webkit-clip-path: polygon(0% -12%, 0% -12%, 169% 63%, 169% 63%);
    clip-path: polygon(0% -12%, 0% -12%, 169% 63%, 169% 63%);
  }

  100% {
    -webkit-clip-path: polygon(0% -12%, 0% 308%, 169% 383%, 169% 63%);
    clip-path: polygon(0% -12%, 0% 308%, 169% 383%, 169% 63%);
  }
}

@keyframes animation-1 {
  0%, 71% {
    -webkit-clip-path: polygon(0% 0%, 0% 78.5%, 100% 34.5%, 100% -44%);
    clip-path: polygon(0% 0%, 0% 78.5%, 100% 34.5%, 100% -44%);
  }

  96%, 100% {
    -webkit-clip-path: polygon(0% 0%, 0% 140%, 100% 96%, 100% -44%);
    clip-path: polygon(0% 0%, 0% 140%, 100% 96%, 100% -44%);
  }
}

@keyframes animation-2 {
  100% {
    stroke-dashoffset: 0;
  }
}

.emotion-0 {
  -webkit-animation: animation-0 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  animation: animation-0 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  will-change: clip-path;
}

.emotion-1 {
  -webkit-animation: animation-1 cubic-bezier(0, 0, 0.85, 1) 2s infinite alternate;
  animation: animation-1 cubic-bezier(0, 0, 0.85, 1) 2s infinite alternate;
  will-change: clip-path;
}

.emotion-2 {
  -webkit-clip-path: polygon(-0.1% -10%, 169% 65%, -0.1% 139%);
  clip-path: polygon(-0.1% -10%, 169% 65%, -0.1% 139%);
}

.emotion-3 {
  display: block;
  stroke: #FFFFFF;
  overflow: visible;
}

.emotion-4 {
  -webkit-animation: animation-2 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  animation: animation-2 cubic-bezier(0.65, 0, 0.55, 1) 2s infinite alternate;
  stroke-dasharray: 9800;
  stroke-dashoffset: 9800;
  will-change: stroke-dashoffset;
}

<div
  class="emotion-0"
  data-testid="loader-wrapper"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <svg
        aria-label="Inverted Loader example"
        class="emotion-3"
        viewBox="0 0 2640 3444"
      >
        <path
          class="emotion-4"
          d="M0 0 M2569 1056L143 2447V149l1175 673v1867l1248 715"
          fill="none"
          stroke-linejoin="round"
          stroke-width="300"
        />
      </svg>
    </div>
  </div>
</div>
`;
